dassh/read_input.py

```diff
@@ -1924,21 +1952,38 @@ def load_materials(self):
                                        temperature=inlet_temp,
                                        from_file=path)
                 else:
-                    # correlation coeffs specified as lists
-                    # Filter None values out of dict
-                    c = {k: v for k, v in self.data['Materials'][m].items()
-                         if v is not None}
-                    matdict[m.lower()] = \
-                        dassh.Material(m.lower(),
-                                       temperature=inlet_temp,
-                                       coeff_dict=c)
+                    if isinstance(self.data['Materials'][m]['density'], str):
+                        # correlation coeffs specified as dict
+                        c = {k: v for k, v in self.data['Materials'][m].items()
```

Comment:
Since this instruction is the same for both the if statement branches, I'd move it before the if statement.